packages/examples/packages/ethereum-provider/src/index.ts

```diff
@@ -207,6 +223,9 @@ async function signTypedData(message: string, from: string) {
  * @see https://docs.metamask.io/snaps/reference/rpc-api/#wallet_invokesnap
  */
 export const onRpcRequest: OnRpcRequestHandler = async ({ request }) => {
+  const { chainId = '0x1' } = (request.params as BaseParams) ?? {};
+  await switchChain(chainId);
```

Comment:
probably not a problem, but i think always using mainnet makes it possible in the extension e2e tests that use this snap to try to hit the mainnet infura endpoint, but I think that's probably only the case if something triggers `getGasPrice` and i'm not certain there is an extension e2e test that does that. Just pointing this out in case I am wrong